src/OptimizerChainFactory.php

```diff
@@ -40,13 +40,20 @@ protected static function getLogger($config): LoggerInterface
     protected static function getOptimizers(array $config)
     {
         return collect($config['optimizers'])
-            ->mapWithKeys(function (array $options, string $optimizerClass) {
-                if (! is_a($optimizerClass, Optimizer::class, true)) {
-                    throw InvalidConfiguration::notAnOptimizer($optimizerClass);
-                }
-
-                return [$optimizerClass => (new $optimizerClass)->setOptions($options)];
-            })
-            ->toArray();
+          ->mapWithKeys(function (array $options, string $optimizerClass) use($config) {
+              if (! is_a($optimizerClass, Optimizer::class, true)) {
+                  throw InvalidConfiguration::notAnOptimizer($optimizerClass);
+              }
+
+              // Initialize optimizer class
+              $newOptimizerClass = new $optimizerClass();
+              if(isset($config['binary_path']) && $config['binary_path']) {
```

Comment:
Move this to its dedicated method. That method should return an empty string if the config option is not set.

Reply:
Just addressed this issue with the newest commit. 

https://github.com/spatie/laravel-image-optimizer/pull/49/commits/c8245f5fc47fb2cb286c06ec0de4ce0f3f87640d

Can you please check?

Thanks: